nrf-hal-common/src/uarte.rs

```diff
@@ -135,31 +201,26 @@ where
             unsafe { w.bits(1) });
 
         // Wait for transmission to end.
-        let mut endtx;
-        let mut txstopped;
-        loop {
-            endtx = self.0.events_endtx.read().bits() != 0;
-            txstopped = self.0.events_txstopped.read().bits() != 0;
-            if endtx || txstopped {
-                break;
-            }
+        while self.0.events_endtx.read().bits() == 0 {
+            // TODO: Do something here which uses less power. Like `wfi`.
         }
 
         // Conservative compiler fence to prevent optimizations that do not
         // take in to account actions by DMA. The fence has been placed here,
         // after all possible DMA actions have completed.
         compiler_fence(SeqCst);
 
-        if txstopped {
-            return Err(Error::Transmit);
-        }
-
         // Lower power consumption by disabling the transmitter once we're
         // finished.
         self.0.tasks_stoptx.write(|w|
             // `1` is a valid value to write to task registers.
             unsafe { w.bits(1) });
 
+        // Wait for transmitter to stop.
```

Comment:
The `tasks_stoptx` write and this spin is unneeded. If you don't abort a tx, just receiving endtx already means the tx machinery is fully stopped.

Reply:
According to Nordic:

```
            // Transmitter has to be stopped by triggering the STOPTX task to achieve
            // the lowest possible level of the UARTE power consumption.
```

Reply:
Not that we seem to care about power consumption (as we busy-wait during a DMA transmit to the UART)

Reply:
Oh huh, strange. When I experimented with this, I saw UARTE using power as long as it's enabled, no matter what the tx/rx state is, so to have low power you have to disable it anyway... Embassy only waits for txstopped when disabling it, and only if the tx was aborted, which seems to work fine.

Seeing nordic has this code I'm now OK with leaving this in, though I don't fully get what the purpose is.